experiments/AMIP/moist_mpi_earth/coupler_utils/ncep_visualizer.jl

```diff
@@ -0,0 +1,137 @@
+"""
+    ncep_paperplots(
+        plot_spec,
+        files_dir;
+        month_date = Dates.DateTime(1979, 01, 01),
+        output_dir = ".",
+        fig_name = "ncep_paperplots",
+    )
+
+Coordinates the postprocessing and plotting of sample fields (specified in `plot_spec`) 
+of a particular monthly mean dataset (specified by `month_date`). Any plot NCEP- specific
+customization should be done here. 
+"""
+function ncep_paperplots(
+    plot_spec,
+    files_dir;
+    month_date = Dates.DateTime(1979, 01, 01),
+    output_dir = ".",
+    fig_name = "ncep_paperplots",
+)
+
+    @info month_date
+
+    tmp_dir = joinpath(files_dir, "ncep_tmp")
+    isdir(tmp_dir) ? nothing : mkpath(tmp_dir)
+
+    ncep_src = NCEPMonthlyDataSource(tmp_dir, [month_date])
+    diags_vnames = propertynames(plot_spec)
+
+    all_plots = []
+    for vname in diags_vnames
+        @info vname
+
+        # download and read data of this month
+        diag_data = get_var(ncep_src, Val(vname))
+
+        # postprocess
+        post_data = postprocess(vname, diag_data, getproperty(plot_spec, vname))
+
+        # create individual plots
+        zonal_mean_params = (; ylabel = "p (hPa)", yaxis = (;), yflip = true) # overwrite defaults
+        p = plot(post_data, zmd_params = zonal_mean_params)
+
+        push!(all_plots, p)
+    end
+
+    # combine plots and save figure
+    save_fig = Plots.plot(
+        all_plots...,
+        size = (1500, 800),
+        right_margin = 12Plots.mm,
+        left_margin = 12Plots.mm,
+        bottom_margin = 12Plots.mm,
+        top_margin = 12Plots.mm,
+    )
+
+    Plots.png(save_fig, joinpath(output_dir, fig_name * ".png"))
+
+    return all_plots
+end
+
+using Downloads
+using NCDatasets
+abstract type DataSource end
+struct NCEPMonthlyDataSource <: DataSource
+    tmp_dir::String
+    month_date::Array
+end
+
+"""
+    download_read_nc(https, tmp_dir, ncep_vname )
+
+Downloads and reads nc datafile of a specified NCEP variable
+"""
+function download_read_nc(ns::NCEPMonthlyDataSource, https, ncep_vname)
```

Comment:
Is `ns` a common abbre for NCEP data? If not, I would use something more explicit.

Reply:
it stands for `NCEPdata source` but i can change it to `data_source` to be more explicit.